pkg/cloudproviders/aws/cred_manager.go

```diff
@@ -0,0 +1,31 @@
+package aws
+
+import (
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+)
+
+// CredentialsManager manages AWS credentials based on the environment.
+type CredentialsManager interface {
+	Start() error
+	Stop()
+	NewSession(cfgs ...*aws.Config) (*session.Session, error)
+}
+
+// DefaultCredentialsManager always returns the default AWS credential chain.
+type DefaultCredentialsManager struct{}
```

Comment:
Same comment as the other PR to use a function instead of exposing a struct type.